### DISCUSSION TITLE
R语言控制台中文乱码问题

### DISCUSSION THREAD

**Discussion by Renmr488** [ShixiangWang/self-study#73] [Q&A] [answered]
您好，请教一个问题，中文在R语言数据库中显示正常，但是做分析时，中文输出在console的结果显示"\u65e5\u5e38\u4ea4\u8c08"之类的乱码（如下），英文和数字显示正常。请问您知道是什么原因吗？

![微信图片_20210329013227](https://user-images.githubusercontent.com/62462250/112847655-dd4c4c00-90d9-11eb-9ef4-449bac687d17.png)
Call:
glm(formula = test ~ relevel(接触方式, "\u65e5\u5e38\u4ea4\u8c08") + 
    relevel(病例年龄组, "<20y") + relevel(病例类型, 
    "\u9633\u6027\u68c0\u6d4b") + relevel(密接性别, "\u7537\u6027") , 
    family = binomial(), data = contact)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-0.9506  -0.3904  -0.1954  -0.0831   3.7869  

Coefficients:
                                                                                                                        Estimate
(Intercept)                                                                                                             -6.04410
relevel(\u63a5\u89e6\u65b9\u5f0f1, "\\u65e5\\u5e38\\u4ea4\\u8c08")\u591a\u79cd\u63a5\u89e6\u65b9\u5f0f                   2.18534
relevel(\u63a5\u89e6\u65b9\u5f0f1, "\\u65e5\\u5e38\\u4ea4\\u8c08")\u5171\u540c\u5c45\u4f4f\u751f\u6d3b                   1.87561
relevel(\u63a5\u89e6\u65b9\u5f0f1, "\\u65e5\\u5e38\\u4ea4\\u8c08")\u805a\u9910                                           1.49519
relevel(\u63a5\u89e6\u65b9\u5f0f1, "\\u65e5\\u5e38\\u4ea4\\u8c08")\u5176\u4ed6                                          -1.70977
relevel(\u75c5\u4f8b\u5e74\u9f84\u7ec4, "<20y")≥60y                                                                     -0.07378
relevel(\u75c5\u4f8b\u5e74\u9f84\u7ec4, "<20y")20-59y                                                                    0.09602

**Comment by ShixiangWang** [comment]
你的 R 版本是多少？是最新版本吗？这个可能是由于 R 本身的原因造成的。

**Reply by Renmr488** [reply]
我的R版本是最新版4.0.4，以前没有这个问题，就是最近才出现。任何数据库都有这个问题

**Comment by ShixiangWang** [comment]
你看看 https://stackoverflow.com/questions/66341119/output-problem-with-chinese-characters-in-r-console-r-version-4，更新到 R4.0.4 应该就可以解决问题了。

**Comment by ShixiangWang** [comment, marked as answer]
那就奇怪了，我最近看官方发布声明说 4.0.4 修复了该问题。你最近有什么异常的操作吗？包括装一些比较奇怪的包，设置 R 的一些配置文件？

另外可以尝试安装最新 https://cran.r-project.org/bin/windows/base/rpatched.html 的滚动版本。如果还是不行，建议完全清除 R 重装。

**Reply by Renmr488** [reply]
多谢指导，重新安装了4.0.5最新版，问题已解决。
